fanficfare/adapters/adapter_silmarillionwritersguildorg.py

```diff
@@ -0,0 +1,236 @@
+        
+        if len(chapters)==1:
+            self.add_chapter(self.story.getMetadata('title'),'https://'+self.host+'/archive/home/'+chapters[0]['href'])
+        else:
+            for chapter in chapters:
+                logger.debug("Added Chapter: "+chapter.string)
+                self.add_chapter(chapter,'https://'+self.host+'/archive/home/'+chapter['href'])
+                
+        
+
+	# find the details section for the work
+	
+        workDetails = soup.find('div', {'id' : 'general'}).find('div', {'id' : 'general'})
+        
+        # Find the summary 
+        summary = workDetails.find_all('p')[3]
+        summary.name='div' # change td to div.  Makes Calibre
+                           # sanitize_html() happier when description
+                           # is empty.
+        self.setDescription(url,summary)
+        
+        # no convenient way to extract metadata so bodge it by finding relevant identifier string and using next element as the data source
+
+        #get characters
+        charList = workDetails.findAll('a', href=re.compile(r'browse.php?type=characters'+"&charid=\d+$"))
+        charText = [char.string for char in charList]
+        self.chararacters = ', '.join(charText)
+        for char in charText:
+            self.story.addToList('characters',char.string)
+            
+        #get warnings
+        warnList = workDetails.findAll('a', href=re.compile(r'browse.php?type=characters'+"&charid=\d+$"))
+        warnText = [warn.string for warn in warnList]
+        self.warnings = ', '.join(charText)
+        for warn in charText:
+            self.story.addToList('warnings',char.string)
+            
+        #get genres
+        genresList = workDetails.findAll('a', href=re.compile(r'browse.php?type=class&type_id=1'+"&classid=\d+$"))
+        genrestext = [genre.string for genre in genresList]
+        self.genre = ', '.join(genrestext)
+        for genre in genrestext:
+            self.story.addToList('genre',genre.string)            
+        
+        #get rating
+        rating = workDetails.find('strong',text='Rated:').next_sibling.string
+        
+        #get completion status and correct for consistency with other adapters
+        if (workDetails.find('strong',text='Completed:').next_sibling.string).lower() == "yes":
+            status="Completed"
+        else:
+            status="In-Progress"
+        
+        #get wordcount
+        wordCount = workDetails.find('strong',text='Word count:').next_sibling.string
+        
+        #get published date, this works for some reason yet doesn't without the spaces in it
+        datePublished = workDetails.find('strong',text=' Published: ').next_sibling.string
+        
+        #get updated date
+        dateUpdated = workDetails.find('strong',text='Updated:').next_sibling.string
+        
+        # try setting metadata values on story, warn if it fails
+        try:
+            self.story.setMetadata('rating', rating)
```

Comment:
This seems like a less than ideal way to handle any errors with these.
1. Errors are much more likely during parsing than during setting.
1. If there was an error in setting an earlier value (_rating_ say), it would prevent the later values(_dateUpdated_, etc) being set, even if they were fine.

Reply:
I've re-factored this to have an individual try except block for each piece of metadata, and have moved the setting of values into their corresponding try block